### PR TITLE
Created new project structure and reproduced plots from tutorial

### PR DESCRIPTION
**Pull request by 0xfabi**
I've recreated the project structure by following the rules from CITEC Data Policy.
All old files and never used files are in the "TODO_remove_for_clean_project". Before publishing we should delete that folder.

### CODE REVIEW

File: src/visualisation_scripts/config/config.yaml [hunk 1]
```diff
@@ -0,0 +1,166 @@
+i/o:
+    # set up your input_path for the input_files
+    input_path: '/Users/fherrmann/Documents/Conquaire/ETACE/data'
```
Comment (svaksha) [inline src/visualisation_scripts/config/config.yaml]: Hi, please dont hardcode paths. Thanks.

File: src/visualisation_scripts/config/config.yaml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    input_files:
+        CentralBank: CentralBank.h5 # please name the key as the agent name
+        Eurostat: Eurostat.h5
+        Firm: Firm.h5
+
+    # set up your output_path
+    output_path: '/Users/fherrmann/Documents/Conquaire/ETACE/results'
```
Comment (svaksha) [inline src/visualisation_scripts/config/config.yaml]: Hi, as above, please dont hardcode paths. Thanks.

File: src/visualisation_scripts/transform.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            return df_out.rename(columns = col_d)
+
+        f_mapper = {'mean': mean, 'sum': summation} # map the function to apply with the desired input
+        return f_mapper[fn]()
+
+
+
+if __name__ == "__main__":
+
+    C = transform_configuration('/home/susupta/Desktop/Fix_preprocess/visualize/src')
```
Comment (svaksha) [inline src/visualisation_scripts/transform.py]: @0xfabi, this hard-coded path will throw errors too.

**Comment by svaksha** [comment]
Hi @0xfabi,  For this project the data files (*.h5) are contained within the main directory (ETACE, FLAViz in future), so dynamic paths in the yaml files makes the code more portable and its easier to use/ maintain. 

Here are some suggestions and helpful links:

*  If there are multiple config files for one project, they should most likely be side-by-side in the root of the project. 
* All paths in the config should be relative to the cwd, else when the file structure is changed all the paths would have to be manually edited, which is sub-optimal.
* If there are various config files for multiple programs, they can be grouped together in a subdirectory - ofcourse, this depends on the program structure and folder structure.  If the directory structure is too deep this should be simplified - nested dirs more than 3-levels deep are a pain. 
* https://medium.com/@ageitgey/python-3-quick-tip-the-easy-way-to-deal-with-file-paths-on-windows-mac-and-linux-11a072b58d5f, suggests solutions for windows users too :+1: 
* Argparse lib (was suggested as an option in this thread): https://stackoverflow.com/questions/45507805/methods-to-avoid-hard-coding-file-paths-in-python

**Comment by svdhoog** [comment]
@0xfabi @svaksha 
Regarding the use of relative paths: 
- Using relative paths for the test data is the way to go, since we ship that data with the release.
- But for the regular use of the library, would relative paths really be that useful? A user may want to completely keep the data and output separate from the location of the library source files. So then, the full path would be listed in the Configuration files, since that is why we have those files in the first place, right? 
- A proposal: could we check if the input path/output path string in the Configuration file starts with "/" and then use the string as a full path, if it does not start with a slash, consider it to be a relative path.
- Note: we do not envision support for Windows.

**Comment by 0xfabi** [comment]
Thanks for your feedback. I've added the possibility to use relative and absolute paths in the config.yaml. The relative paths start inside the project folder. You use absolute paths by starting with a "/".

**Comment by svaksha** [comment]
On Thu, Apr 12, 2018 at 3:12 PM, svdhoog <notifications@github.com> wrote:
>
> @0xfabi @svaksha
> Regarding the use of relative paths:
>
> Using relative paths for the test data is the way to go, since we ship that data with the release.

I agree.

> But for the regular use of the library, would relative paths really be that useful? A user may want to completely keep the data and output separate from the location of the library source files. So then, the full path would be listed in the Configuration files, since that is why we have those files in the first place, right?

The dynamic path suggestion was for the data IO and programs that the
user invokes. For output storage, the user should have the freedom to
define the storage path.

> Note: we do not envision support for Windows.

Ok, now I have a better understanding of what you meant by "packaging"
:-) The folksonomy for identical terms used outside the tech world is
what led to this confusion

**Pull request merged**